### PR TITLE
avoid failing on wrapped errors

### PR DESCRIPTION
**Pull request by uturunku1**
## Description

I noticed logreader.go was using the old way of comparing errors `==`. This make sense because this file was implemented almost 4 years ago when `errors.Is` was not an available standardize practice yet: https://go.dev/blog/go1.13-errors

This could potentially fail an assertion if the error thrown by that external package had wrapped errors. So I change this implementation to use `errors.Is`.

The other small change I am introducing here is that instead of returning a semi-vague error thrown by an external library like so:

```
file, err := os.Stat(path)
	if err != nil {
            return err
```

I am unwrapping the error, adding additional information about the call site where things failed, and then wrapping back the error with `fmt.Errorf(...)`, like this:

```
if err != nil {
		if errors.Is(err, fs.ErrNotExist) {
			return fmt.Errorf(`failed to find terraform configuration files under the path "%v": %w`, path, err)
		}
		return fmt.Errorf(`unable to upload terraform configuration files from the path "%v": %w`, path, err)
	}
``` 

Adding some context to an error coming from a large external package, should make identifying the source of problems easier during debugging.

## Output from tests (HashiCorp employees only)

```
$ TFE_ADDRESS="https://example" TFE_TOKEN="example" TF_ACC="1" go test ./... -v -tags=integration -run TestFunctionsAffectedByChange

...
```

### CODE REVIEW

**Review by sebasslash — APPROVED** [review]
👍

**Comment by sebasslash** [comment]
Disregard, I didn't notice it was a draft.

File: tfe.go [hunk 1]
```diff
@@ -870,8 +872,8 @@ func decodeErrorPayload(r *http.Response) ([]string, error) {
 	return errs, nil
 }
 
-func errorPayloadContains(errors []string, match string) bool {
-	for _, e := range errors {
+func errorPayloadContains(payloadErrors []string, match string) bool {
```
Comment (uturunku1) [inline tfe.go:875]: unrelated change to this PR, fyi. Golangci-lint warned me that the previous name assigned to this variable was clashing with the name of the external library "errors".

**Review by sebasslash — APPROVED** [review]
👍

**Comment by sebasslash** [comment]
I wasn't aware of this. Read up on `errors.Is()`, very cool -- I'd been comparing errors using `==` all this time. I'm wondering if the tfe provider might need this change. 🤔
